### ISSUE TITLE
Why is glob-watcher a dependency of vinyl-fs?

### ISSUE THREAD

**Issue by RnbWd** [wearefractal/vinyl-fs#54]
I'm just wondering. In gulp it's used to watch tasks, but it's not passed directly like `vfs.src` or `vfs.dest`:

```js
Gulp.prototype.src = vfs.src;
Gulp.prototype.dest = vfs.dest;
Gulp.prototype.watch = function (glob, opt, fn) {
  if (typeof opt === 'function' || Array.isArray(opt)) {
    fn = opt;
    opt = null;
  }

  // array of tasks given
  if (Array.isArray(fn)) {
    return vfs.watch(glob, opt, function () {
      this.start.apply(this, fn);
    }.bind(this));
  }
  return vfs.watch(glob, opt, fn);
};
```

I'm using vinyl-fs in a project, but I don't use glob-watcher, so I forked the repo without it (to save 10 seconds when npm installing). This is definitely a minor issue, but I see it as an erroneous dependency. When I rebuilt 'the machine', I realized that watching files should be the task manager's responsibility. I use globby + chokidar, it works great on my mac-os, but I'm not maintaining gulp, and I don't have thousands of users to worry about. I don't know the pros and cons of gaze vs. chokidar, but I what I do know is that gaze was last published 9 months ago, while chokidar was published 9 days ago. (not trying to say that publish date is more important than consistency, but I've noticed improvements on my system)

**Comment by phated** [comment]
watch uses glob-watcher

**Comment by contra** [comment]
Can you clarify what you're asking here? Your post covers a lot:

1. Why does gulp augment vinyl-fs.watch
2. Why does vinyl-fs.watch exist
3. Why do we use gaze instead of chokidar


Answers:

1. Because it adds the ability to specify an array instead of a function to run tasks, which is gulp-specific
2. Because when I came up with the idea of a vinyl-adapter the three things I thought of were src, dest, and watch which are store-dependent (s3 watch will be different from fs watch, so the vinyl-adapter will handle that)
3. Because gaze handles globs and chokidar doesnt, but I've opened an issue on chokidar and they seem open to adding this

**Comment by sindresorhus** [comment]
> gaze handles globs and chokidar doesnt, but I've opened an issue on chokidar and they seem open to adding this

Does that really matter though. It's easy to just resolve the globs before passing the file list.

**Comment by contra** [comment]
@sindresorhus And when a new file or folder is added?

**Comment by sindresorhus** [comment]
Hmm, didn't consider that behaviour. But looks like Chokidar is close to support globbing ;)

**Comment by RnbWd** [comment]
@contra - thanks for the responses! My question was mostly centered about 'why vinyl-fs.watch' exists, which you answered in #2. 

@sindresourhus also makes a great point, and I wrote a gist that shows (on a basic level) how I implemented it - https://gist.github.com/RnbWd/21cd37654b6926e3043b

It's not well tested at this point, certainly not cross platform, but it works well on my system. So I felt compared to share. Most file-watcher in nodjes have a tendency to be  inconsistent across platforms or how they handle large file-sets, so there's definitely not a one-size-fits all approach for all use cases.
